### PR TITLE
Refactor test suites

### PR DESCRIPTION
**Pull request by aleochoam**
### Summary
- Added the missing database-cleaner gem
- Removed the TODO comment on a test
- Refactor the controller specs with the comments suggested
- Added the missing test for some models

### CODE REVIEW

File: app/controllers/api/v1/rents_controller.rb [hunk 1]
```diff
@@ -6,7 +6,9 @@ def create
         authorize rent
 
         if rent.save
-          RentMailer.with(rent_id: rent.id).rent_confirmation.deliver_later
+          RentMailer.with(rent_id: rent.id).rent_confirmation.deliver_later unless
+            params[:send_email]
```
Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: Why did you add this a params? It's because each time you run you test you receive an email? 
So this solution is not the good one.
In your `config/environment/test.rb` you should have this config:
`config.action_mailer.delivery_method = :test`

File: spec/controllers/rent_controller_spec.rb [hunk 2]
```diff
@@ -46,20 +46,31 @@
   end
 
   describe 'POST #create' do
-    subject(:response) do
+    subject(:http_request) do
       post :create, params: {
         user_id: user.id,
-        rent: attributes_for(:rent, user_id: user.id)
+        rent: attributes_for(:rent, user_id: user.id, book_id: book.id),
+        send_email: false
```
Comment (Gmarn) [inline spec/controllers/rent_controller_spec.rb]: Remove this params ;)

File: spec/controllers/rent_controller_spec.rb [hunk 3]
```diff
@@ -46,20 +46,31 @@
...
     end
 
+    let(:book) { create(:book) }
+
     context 'with authentication' do
       include_context 'authenticated user'
 
-      # TODO: see how to handle the no redis server error
-      xit do
+      it do
```
Comment (Gmarn) [inline spec/controllers/rent_controller_spec.rb]: Add a description to your `it`

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -62,4 +62,16 @@
   config.include FactoryBot::Syntax::Methods
   config.include Devise::Test::ControllerHelpers, type: :controller
   config.include Response::JSONParser
+
```
Comment (Gmarn) [inline spec/rails_helper.rb]: Ok, but better to put this config in your `spec_helper.rb`

Also here the config we use in the [rails bootstrap](https://github.com/Wolox/rails-bootstrap/blob/master/spec/support/spec_helper.rb)

File: app/controllers/api/v1/rents_controller.rb (hunk 1 above)
Comment (aleochoam) [reply app/controllers/api/v1/rents_controller.rb]: The problem i have is that an `Redis::CannotConnectError` is thrown,  and i cant figure it how to catch it

Comment (Gmarn) [reply app/controllers/api/v1/rents_controller.rb]: But why did you not put this config ?
`config.action_mailer.delivery_method = :test`

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -93,4 +94,15 @@
   # as the one that triggered the failure.
   Kernel.srand config.seed
 =end
+
+  config.before(:suite) do
+    DatabaseCleaner.strategy = :transaction
+    DatabaseCleaner.clean_with(:truncation)
+  end
+
+  config.around(:each) do |example|
```
Comment (Gmarn) [inline spec/spec_helper.rb]: Humm, did this do the same job than:
```
  config.before(:each) do
    DatabaseCleaner.start
  end

  config.after(:each) do
    DatabaseCleaner.clean
  end
```

Comment (Gmarn) [reply spec/spec_helper.rb]: ?

Comment (aleochoam) [reply spec/spec_helper.rb]: Yes!

https://github.com/DatabaseCleaner/database_cleaner#rspec-example
https://github.com/DatabaseCleaner/database_cleaner#how-to-use
